File: python/silapay_connector/silapay_connector/providers/adyen.py
import os


class AdyenProvider:
    def __init__(self, api_key=None, merchant_account=None):
        self.api_key = api_key or os.getenv("ADYEN_API_KEY")
        self.merchant_account = merchant_account or os.getenv("ADYEN_MERCHANT_ACCOUNT")

    def pix(self, dados):
        raise NotImplementedError(
            "Adyen: metodo pix() ainda nao implementado. Consulte https://docs.adyen.com/"
        )

    def boleto(self, dados):
        raise NotImplementedError(
            "Adyen: metodo boleto() ainda nao implementado. Consulte https://docs.adyen.com/"
        )

    def cartao(self, dados):
        raise NotImplementedError(
            "Adyen: metodo cartao() ainda nao implementado. Consulte https://docs.adyen.com/"
        )

    def saldo(self):
        raise NotImplementedError(
            "Adyen: metodo saldo() ainda nao implementado. Consulte https://docs.adyen.com/"
        )

File: typescript/silapay-connector/src/providers/adyen.ts
import type { ConnectorOptions, PaymentProvider } from "../types";

export class AdyenProvider implements PaymentProvider {
    private readonly apiKey?: string;
    private readonly merchantAccount?: string;

    constructor(options: ConnectorOptions = {}) {
        this.apiKey = options.apiKey || process.env.ADYEN_API_KEY;
        this.merchantAccount =
            options.merchantAccount || process.env.ADYEN_MERCHANT_ACCOUNT;
    }

    async pix(): Promise<never> {
        throw new Error("Adyen: método pix() ainda não implementado. Consulte https://docs.adyen.com/");
    }

    async boleto(): Promise<never> {
        throw new Error(
            "Adyen: método boleto() ainda não implementado. Consulte https://docs.adyen.com/"
        );
    }

    async cartao(): Promise<never> {
        throw new Error(
            "Adyen: método cartao() ainda não implementado. Consulte https://docs.adyen.com/"
        );
    }

    async saldo(): Promise<never> {
        throw new Error("Adyen: método saldo() ainda não implementado. Consulte https://docs.adyen.com/");
    }
}
